src/scripts/object.search.js

```diff
@@ -0,0 +1,210 @@
+// Description:
+//	Always listening, waiting to translate
+//
+// Configuration:
+// HUBOT_OBJECT_STORAGE_AUTH_URL The Auth URL provided by Object Storage service
+// HUBOT_OBJECT_STORAGE_USER_ID The Object Storage user id
+// HUBOT_OBJECT_STORAGE_PASSWORD The Objeect Storage password
+// HUBOT_OBJECT_STORAGE_PROJECT_ID The Object Storage project id
+// HUBOT_OBJECT_STORAGE_BLUEMIX_REGION The bluemix region you wish to target.  For example dallas.
+//
+// Author:
+//  @kholdaway
+//
+
+/*
+ * Licensed Materials - Property of IBM
+ * (C) Copyright IBM Corp. 2016. All Rights Reserved.
+ * US Government Users Restricted Rights - Use, duplication or
+ * disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+ */
+'use strict';
+
+const env = require('../lib/env');
+const Helper = require('../lib/paramHelper');
+const path = require('path');
+const TAG = path.basename(__filename);
+const Conversation = require('hubot-conversation');
+const _ = require('lodash');
+const activity = require('hubot-ibmcloud-activity-emitter');
+const NLCHelper = require('../lib/nlcHelper');
+
+const i18n = new (require('i18n-2'))({
+	locales: ['en'],
+	extension: '.json',
+	// Add more languages to the list of locales when the files are created.
+	directory: __dirname + '/../messages',
+	defaultLocale: 'en',
+	// Prevent messages file from being overwritten in error conditions (like poor JSON).
+	updateFiles: false
+});
+// At some point we need to toggle this setting based on some user input.
+i18n.setLocale('en');
+
+let helper;
+let storage;
+module.exports = (robot, res) => {
+	if (!helper) {
+		helper = new Helper({
+			robot: robot,
+			res: res,
+			settings: env
+		});
+		if (helper.initializedSuccessfully()) {
+			storage = helper.getObjectStorage();
+		}
+		else {
+			storage = undefined;
+		}
+	}
+
+	const switchBoard = new Conversation(robot);
+	let context = {
+		res: res,
+		robot: robot,
+		switchBoard: switchBoard,
+		settings: env
+	};
+
+	let nlcHelper = new NLCHelper(context);
+
+	enableClassifierCleanupInterval();
+
+	// Natural Language match
+	robot.on('objectstorage.search.object', (res, parameters) => {
+		robot.logger.debug(`${TAG}: Natural Language match.`);
+		// Parameter values are obtain through the cognitive/nlc process.
+		// Verify that required parameter values were succesfully obtained.
+		if (parameters && parameters.searchphrase) {
+			processObjectSearch(robot, res, parameters.searchphrase);
+		}
+		else {
+			robot.logger.error(`${TAG}: Error extracting container name from text [${res.message.text}].`);
```

Comment:
This error print out should should say "Error extracting search phrase from text...", not container name.